scripts/bump.sh

```diff
@@ -1,56 +1,97 @@
 #!/bin/sh
 
-# ----------- Basic Checking -----------
-
-# Check that 1 arg has been supplied
-if [ "$#" != "1" ]; then
-	echo "[-] Exactly one argument is needed."
-	echo '[?] Example: `./bump.sh 0.2.6`'
-	exit 1
-fi
-
 dependencies=(
 	semver
 	git
 	npm
 )
 
-# Check that required binaries are installed
-for package in ${dependencies[@]}; do
-	if [ ! $(which $package) ]; then
-		echo "[-] All of '${dependencies[@]}' are needed to bump version"
+# ----------- Basic Checking -----------
+
+# check that current directory is repo root
+check_pwd () {
+	root="$(dirname $(realpath $(dirname "$0")))"
+	[ "$root" != "$(pwd)" ] && {
+		echo "[!] the script should be run from the root of the repo"
+		echo "[-] expected '$root'"
+		echo "[-] found '$(pwd)'"
+		exit 1
+	}
+	[ -z "$DRY_RUN" ] || echo "[i] Current directory is repo's root OK"
+}
+
+# Check that 1 arg has been supplied
+check_arg () {
+	if [ "${NB_ARGS}" != "1" ]; then
+		echo "[-] Exactly one argument is needed, got ${NB_ARGS}."
+		echo '[?] Example: `./bump.sh 0.2.6`'
 		exit 1
 	fi
-done
+	[ -z "$DRY_RUN" ] || echo "[i] Only argument has been given OK"
+}
+
+# Check that required binaries are installed
+check_dependencies () {
+	for package in ${dependencies[@]}; do
+		if [ ! $(which $package) ]; then
+			echo "[-] All of '${dependencies[@]}' are needed to bump version"
+			exit 1
+		fi
+	done
+	[ -z "$DRY_RUN" ] || echo "[i] Dependencies: '${dependencies[@]}' are installed on the system OK"
+}
 
 # Check that supplied version is semantically correct
-if [ ! "$1" = "$(semver $1)" ]; then
-	echo "[-] $1 is not a valid version number according to semver"
-	exit 1
-fi
+check_version_semantics () {
+	if [ ! "${NEW_VERSION}" = "$(semver ${NEW_VERSION})" ]; then
+		echo "[-] ${NEW_VERSION} is not a valid version number according to semver"
+		exit 1
+	fi
+	[ -z "$DRY_RUN" ] || echo "[i] Provided version '${NEW_VERSION}' is semantically correct OK"
+}
 
 # Check that git working directory is clean...
-if [ -n "$(git status --short --untracked-files=no)" ]; then
-	echo "[-] git working directory isn't clean"
-	exit 1
-fi
+check_clean_git_working_dir () {
+	if [ -n "$(git status --short --untracked-files=no)" ]; then
+		echo "[-] git working directory isn't clean"
+		exit 1
+	fi
+	[ -z "$DRY_RUN" ] || echo "[i] git working directory is clean OK"
+}
+
+# Variables
+NB_ARGS=$#
```

Comment:
maybe slightly out of the scope of this PR, as some of the lines i will talk about did only move in these changes, so tell me if i should open an issue :+1: 

i think
```bash
NB_ARGS="$#"
```
would be better

it does not change that much on that line, but when there is user input involved, it's always good to wrap `$1`, `$2`, ..., in quotes :+1: 

it avoids passing arguments and options to inside functions and commands :confused:
_it's the only CS challenge i've ever done, so i remember exactly the issue here :eyes:_ 

not a huge deal here i think, but i believe it's good pratice to wrap everything in quotes when writing `bash` scripts :wink:

Reply:
Actually you're right but 3 things here:
 - it doesn't change anything here because `$#` can't be fully controlled by the user (it's only a number) but your remark applies for `$1`
 - quoting is a good practice but in fact it is not sufficient you can bypass quoted variables too...
 - here the script hasn't a setuid bit so there isn't any security concerns :slightly_smiling_face: (and I **strongly** advice you to never put the setuid bit to a shell script (at least bash/sh)) 
 
But it's good practice after all so I'll do it!